impl/src/planning.rs

```diff
@@ -461,14 +454,41 @@ impl<'planner> WorkspaceSubplanner<'planner> {
           crate_catalog_entry: &own_crate_catalog_entry,
           source_id: &own_source_id,
           node: &node,
-          crate_settings: &crate_settings,
+          crate_settings: self.crate_settings(&own_package).unwrap(),
           sha256: &checksum_opt.map(|c| c.to_owned()),
         };
 
         Some(crate_subplanner.produce_context())
       })
       .collect()
   }
+
+  fn crate_settings(&self, package: &Package) -> Result<Option<&CrateSettings>> {
+    self
+      .settings
+      .crates
+      .get(&package.name)
+      .map_or(Ok(None), |settings| {
+        let mut versions = settings
+          .iter()
+          .filter(|(ver_req, _)| ver_req.matches(&package.version))
+          .peekable();
+
+        match versions.next() {
+          None => Ok(None),
```

Comment:
In what case would this actually occur?

Looking at the unwrap_or_default on line 535, it seems like this would be an error, but maybe I'm not understanding a particular case.

Reply:
I cant think of how, although I guess its habit for me to handle Option without `unwrap` (because you know code maintenance)

I will poke at it a bit and see if I can figure out if needed

Reply:
This is specific to times where we dont match (no match there is a use the default, but that `unwrap_or_default` highlighted does not provide a default config here)